python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,580 @@
+    ArrayD = FieldFormatter(lambda f: _sqlTypeForArray("d", f), lambda v: _formatArray("d", v)),
+)
+
+
+def canonicalizeFieldName(fieldName):
+    """Return a MySQL-compatible version of the schema field name, for now by
+    changing any non-word characters to underscores.
+    """
+    return re.sub(r"[^\w]", "_", fieldName)
+
+def quoteIdentifier(identifier):
+    """Return a MySQL compatible version of the given identifier by quoting
+    it with back-ticks (and escaping embedded back-ticks).
+    """
+    return "`" + identifier.replace("`", "``") + "`"
+
+def aliasesFor(name, mappings):
+    """Compute and return the set of possible aliases for the given field name.
+
+    The afw table library processes a given name F by replacing the longest
+    prefix of F that can be found in a schema's alias map with the corresponding
+    alias target. Replacement is repeated until either no such prefix is found,
+    or N replacements have been made (where N is the size of the alias map).
+
+    The mappings argument must be a sorted list of substitutions, where a single
+    substitution is a 2-tuple of strings (source, target).
+    """
+    n = 0
+    aliases, names = set(), set()
+    names.add(name)
+    while n < len(mappings) and len(names) > 0:
+        # Perform one round of reverse alias substitution
+        n += 1
+        newNames = set()
+        for name in names:
+            for i, (source, target) in enumerate(mappings):
+                if name.startswith(target):
+                    alias = source + name[len(target):]
+                    # alias is only valid if source is its longest prefix in mappings.
+                    # A prefix strictly longer than source must occur after it in the
+                    # sorted list of mappings.
+                    valid = True
+                    for s, _ in mappings[i + 1:]:
+                        if not s.startswith(source):
+                            break;
+                        if alias.startswith(s):
+                            valid = False
+                            break
+                    if valid:
+                        aliases.add(alias)
+                        newNames.add(alias)
+        names = newNames
+    return aliases
+
+
+class IngestCatalogConfig(pexConfig.Config):
+    """Configuration for the IngestCatalogTask.
+    """
+
+    allowReplace = pexConfig.Field(
+        "Allow replacement of existing rows with the same unique IDs",
+        bool, default=False
+    )
+
+    maxQueryLen = pexConfig.Field(
+        "Maximum length of a query string. None means use a non-standard, "
+        "database-specific way to get the maximum.",
+        int, optional=True, default=None
+    )
+
+    maxColumnLen = pexConfig.Field(
+        "Maximum length of a database column name or alias. Fields "
+        "that map to longer column names will not be ingested.",
+        int, default=64
```

Comment:
I'll sanity check it for now. Making this code backend agnostic might or might not cause this parameter to go away. It depends on whether this code will end up needing to know exactly what it is talking to (in which case it can supply the appropriate value), or whether it will end up using some database abstraction library. In the latter case, if the abstraction layer we use does not provide a value for this type of constraint, I think that providing a user settable parameter to get around failures is a reasonable thing to do.